extensions/wikia/Piggyback/Piggyback_body.php

```diff
@@ -45,89 +46,84 @@ function execute( $par ){
+		$this->templateData['actionlogin'] = $this->titleObj->getLocalUrl( 'action=submitlogin' );
 
 		$this->mOtherName = $request->getVal( 'wpOtherName' );
 		parent::load();
 
-		$this->exTemplate->set("link", "");
-		$this->exTemplate->set("usedomain", "");
-		$this->exTemplate->set("canremember", "");
-		$this->exTemplate->set("name", "");
-		$this->exTemplate->set("useemail", "");
-		$this->exTemplate->set( 'message', "" );
-
 		$this->mType = "login";
 		/* fake to don't change remember password */
 		$this->mRemember = (bool) $wgUser->getOption( 'rememberpassword' );
 	}
 
 	function mainLoginForm( $msg, $msgtype = 'error' ) {
-		$this->exTemplate->set( 'messagetype', $msgtype );
-		$this->exTemplate->set( 'message', $msg );
-		$this->exTemplate->set( 'name', $this->mUsername );
-		$this->exTemplate->set( 'password', $this->mPassword );
-		$this->plugin->set( 'otherName', $this->mOtherName );
+		$this->templateData['messagetype'] = $msgtype;
+		$this->templateData['message'] = $msg;
+		$this->templateData['name'] = $this->mUsername;
+		$this->templateData['password'] = $this->mPassword;
+		$this->templateData['otherName'] = $this->mOtherName;
 	}
 
 	function successfulLogin() {
-		global $wgUser, $wgAuth,$wgOut,$wgRequest;
+		global $wgUser, $wgAuth, $wgOut, $wgRequest;
 
 		/* post valid */
 		$u = User::newFromName( $this->mOtherName );
 
 		$cu = User::newFromName( $this->mUsername );
 
-		if (!$cu->checkPassword( $this->mPassword )) {
-			if( $retval = '' == $this->mPassword ) {
-				$this->mainLoginForm( wfMsg( 'wrongpasswordempty' ) );
+		if ( !$cu->checkPassword( $this->mPassword ) ) {
+			if ( $retval = '' == $this->mPassword ) {
+				$this->mainLoginForm( wfMessage( 'wrongpasswordempty' )->escaped() );
 			} else {
-				$this->mainLoginForm( wfMsg( 'wrongpassword' ) );
+				$this->mainLoginForm( wfMessage( 'wrongpassword' )->escaped() );
 			}
-			return ;
+
+			return;
 		}
 
-		if ( !is_object( $u ) || $u->getId()  == 0 ) {
-			$this->mainLoginForm( wfMsg( 'piggyback-nosuchuser', htmlspecialchars( $this->mOtherName ) ) );
-			return ;
+		if ( !is_object( $u ) || $u->getId() == 0 ) {
+			$this->mainLoginForm( wfMessage( 'piggyback-nosuchuser', $this->mOtherName )->escaped() );
+
+			return;
 		}
 
-		if ( $u->getId()  == $wgUser->getId() ) {
-			$this->mainLoginForm( wfMsg( 'piggyback-itisyou') );
-			return ;
+		if ( $u->getId() == $wgUser->getId() ) {
+			$this->mainLoginForm( wfMessage( 'piggyback-itisyou' )->escaped() );
+
+			return;
 		}
 
 		$wgRequest->setSessionData( "PgParentUser", $wgUser->getID() );
 
-		wfRunHooks( 'PiggybackLogIn',array( $wgUser, $u) );
+		wfRunHooks( 'PiggybackLogIn', array( $wgUser, $u ) );
 
 		$log = new LogPage( 'piggyback' );
-		$log->addEntry( 'piggyback', SpecialPage::getTitleFor( 'Piggyback') , "login ".$wgUser->getName()." to ".$u->getName(),  array() );
+		$log->addEntry( 'piggyback', SpecialPage::getTitleFor( 'Piggyback' ), "login " . $wgUser->getName() . " to " . $u->getName(), array() );
```

Comment:
@michalroszka Is string interpolation enough?
I'm not sure if we need special formatting here.